### DISCUSSION TITLE
Online music database graphs

### DISCUSSION THREAD

**Discussion by username116** [username116/Music-Graph#2] [Show and tell]
**[Discogs graph](https://username116.github.io/Music-Graph/Draw%20Graph%20Discogs.html)**

**[AllMusic graph](https://username116.github.io/Music-Graph/Draw%20Graph%20AllMusic.html)**

**[Rate Your Music graph](https://username116.github.io/Music-Graph/Draw%20Graph%20Rate%20Your%20Music.html)**

**[MusicBrainz graph](https://username116.github.io/Music-Graph/Draw%20Graph%20MusicBrainz.html)**

**Comment by username116** [comment]
Sources:
Discogs genres: the [Add release page](https://www.discogs.com/release/add) (need an account)
[AllMusic genres](https://www.allmusic.com/genres)
[Rate Your Music genres](https://rateyourmusic.com/genres)
[MusicBrainz genres](https://musicbrainz.org/genres)

For the AllMusic graph, I used the list in the [descriptors_xxx_allmusic](https://github.com/username116/Music-Graph/blob/main/main/music_graph/music_graph_descriptors_xxx_allmusic.js) file. I think some categories like 'Avant-Garde', 'Children's' or 'Comedy / Spoken' are missing. Maybe not up to date with the latest changes on this file.

MusicBrainz genre data in csv format (countries, subgenres, fusions, influences, Wikidata items and links to other databases): [MusicBrainz genres.txt](https://github.com/username116/Music-Graph/files/13464994/MusicBrainz.genres.txt)

**Reply by regorxxx** [reply]
Thanks for all this work! Looking great as a first approach.

As said in the other thread:
- The graphs would need influence-subsitutions-style clusters to make them really work as a whole.
- You have to consider if you want a central point, i.e. all genre big clusters being equally separated between them; or some other kind of "minimum" similarity between those clusters. For ex. rock and pop clusters are more similar than Classical and Folk.

If you provide the descriptors for those graphs I could try adding more substitutions (since I would skip the scrapping part).
For the musicBrainz file, I can totally reuse that to add more data to the current graph. Specially for the country part, etc.

EDIT: just saw it at the other branches, can work with that.

**Comment by username116** [comment]
The central point on the first two graphs was to avoid errors in the console, then I discovered the debug = false parameter. Then I left it to do something visual 🌼. But yes, this central point has no musical value. These graphs can be improved.

Otherwise I didn't add anything more. All the other links exist in the online database trees.

> If you provide the descriptors for those graphs I could try adding more substitutions
> just saw it at the other branches, can work with that.

Yes, it's all in the [other-graphs](https://github.com/username116/Music-Graph/tree/other-graphs/main/music_graph) branch. Here are some files with work steps. The result is already online, but if it helps...  [Genres.zip](https://github.com/username116/Music-Graph/files/13497416/Genres.zip)


<details><summary>Details</summary>
</p>

I discovered in the code of the Discogs page that special characters are encoded like this: `Samba-Can\u00e7\u00e3o`  ->  `Samba-Canção`
Apparently it's called JS character encoding/decoding. I used [this](https://bitbucket.org/rdipardo/htmltag/src/default/) to convert it.

In the Rate Your Music tree, identical nodes can be found in several places and at different levels. For example, there are:
```
Folk -> Traditional Folk Music -> European Folk Music -> Celtic Folk Music -> Scottish Folk Music -> ...
Regional Music -> European Music -> British Music -> Scottish Folk Music -> ...
Regional Music -> European Music -> European Folk Music -> Celtic Folk Music -> Scottish Folk Music -> ...
Regional Music -> Traditional Folk Music -> European Folk Music-> Celtic Folk Music -> Scottish Folk Music -> ...
```

At first, I separated the levels, but that made duplicate nodes. That's why I put everything on the same level and removed the duplicate lines. In the end, that's a lot of links in the graph. And if you want to look at `Scottish Folk Music` on [the graph](https://username116.github.io/Music-Graph/Draw%20Graph%20Rate%20Your%20Music.html), it's here:  
<details><summary>Details</summary>
<p>

![Screenshot](https://github.com/username116/Music-Graph/assets/108369600/24fb809d-8697-458d-a51c-04be4dbbe760)

</p>
</details> 

<p>
</details>

**Comment by username116** [comment]
@UltimateRiff, maybe this will interest you, about [LB-1509 - Genre explorer](https://tickets.metabrainz.org/browse/LB-1509). I just copied the genre lists, the code is not mine.

**Reply by UltimateRiff** [reply]
thanks for the heads-up~ thas pretty cool~ I shared it over there for some inspiration~
